sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -132,17 +135,49 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
         if (promises.length >= maxPromisesAllowed) {
           // Waits for all be processed.
           const insertResults = await Promise.all(promises);
+          await this.saveBulkOfferingParentId(
+            insertResults.map((result) => result.createdOfferingId),
+            offeringRepo,
+          );
           allResults.push(...insertResults);
           // Clear the array.
           promises.length = 0;
         }
       }
       const finalResults = await Promise.all(promises);
+      await this.saveBulkOfferingParentId(
+        finalResults.map((result) => result.createdOfferingId),
+        offeringRepo,
+      );
       allResults.push(...finalResults);
       return allResults;
     });
   }
 
+  /**
+   * Update offering parent ids for given offerings.
+   * * When creating a new offering, parent id and
+   * * the primary id are the same.
```

Comment:
Are the double stars intentional?
![image](https://github.com/bcgov/SIMS/assets/78114138/f71bdafd-0db7-46f7-b6f4-4d2e09614c2d)

Reply:
yes @andrepestana-aot, But I was not expecting it to be a bulletin point. I will remove it.